odoo/orm/fields_numeric.py

```diff
@@ -191,6 +191,13 @@ def __init__(self, string: str | Sentinel = SENTINEL, currency_field: str | Sent
     def _description_currency_field(self, env: Environment) -> str | None:
         return self.get_currency_field(env[self.model_name])
 
+    def _description_aggregator(self, env: Environment):
+        model = env[self.model_name]
+        currency_field_name = self.get_currency_field(model)
+        currency_field = model._fields[currency_field_name]
+
+        return currency_field._description_groupable(env) and super()._description_aggregator(env)
```

Comment:
I missed this one: `_description_groupable` doesn't have the semantics. IMO you should check the currency field accept `arrag_agg_distinct` instead. 

```suggestion
    def _description_aggregator(self, env: Environment):
        model = env[self.model_name]
        query = model._as_query(ordered=False)
        currency_field_name = self.get_currency_field(model)
        currency_field = model._fields[currency_field_name]
        # The currency field need to be aggregable too.
        if not currency_field.column_type or not currency_field.store:
            try:
                model._read_group_select(f"{currency_field_name}:array_agg_distinct", query)
            except (ValueError, AccessError):
                return None
        return super()._description_aggregator(env)
```

Everything else looks fine to me.